app.ts

```diff
@@ -0,0 +1,171 @@
+        for (let i = fromID; i <= toID; i++) {
+            let element = ($("#" + i.toString()).offset() as JQueryCoordinates).top as number;
+
+            if ( element >= documentHeight ){
+                return i;
+            } else {
+                continue
+            };
+        }
+        return toID
+}
+
+function previousPageResize(previousCursor: Array<number>): Array<number> {
+    let fromId: number = previousCursor[0] - (nextPageResize(previousCursor) - previousCursor[0]);
+    let toId: number = fromId + step;
+    return [previousCursor[0] - (nextPageResize(previousCursor) - previousCursor[0]), toId]
+}
+//#endregion
+//#endregion
+
+const step: number = 37;
+let previousCursor: Array<number>;
+let previousScale: number;
+
+window.onload = () => {     
+    previousCursor = getPageContent(0,37)
+    
+    $("#previous-page").click(() => { 
+        previousCursor = previousPageResize(previousCursor);
+        let fromId: number = (previousCursor[0] >= 0 ? previousCursor[0] : 0) as number;
+        let toId: number = (previousCursor[0] >= 0 ? previousCursor[1] : 37) as number;
+        getRecordCount()
+            .then((value) => {
+                fromId = fromId == value - 1 ? fromId - step : fromId;
+                toId = toId <= value - 1 ? toId : value - 1;
+                previousCursor = placeRecords(fromId, toId);
+            });
+        
+    });
+
+    $("#next-page").click(() => {
+            let fromId: number = nextPageResize(previousCursor) as number;
+            getRecordCount()
+                .then((value) => {
+                    if ( fromId <= value - step - 1 ) {
+                        let toId: number = fromId + step <= value - 1 ? fromId + step : value - 1;
+                        previousCursor = placeRecords(fromId, toId);
+                    } else if (fromId <= value - 1)  {
+                        previousCursor = placeRecords(fromId, value - 1);
+                    } else {
+                        alert('You have reached the end of the list')
+                    }
+                });        
+        });
+
+    $("#go-to-button").click(() => {
+        let fromId = toNumber($("#go-to-index").val() as string, 2);
+        getRecordCount()
+            .then((value) => {
+                if (Math.floor(fromId).toString() == fromId.toString() === true) {
+                    if ( fromId > value - step ) {
+                        alert('You may not insert a value greater than ' + (value - step).toString());
+                    } else {
+                        let toId: number = (fromId as number) + step < value ? (fromId as number) + step : value - 1;
+                        previousCursor = placeRecords(fromId, toId);
+                    }
+                } else {
+                    alert('It seems you are not inserting an integer - please ensure that you are.');
+                }
+            });
+    });
+}
+let resizeTimer: number = 50;
```

Comment:
I would assign this to -1 or 0 to make it seem a bit more like an "empty" variable.